asynciterator.ts

```diff
@@ -1717,7 +1717,7 @@ export class MultiTransformIterator<S, D = S> extends TransformIterator<S, D> {
   @extends module:asynciterator.BufferedIterator
 */
 export class UnionIterator<T> extends BufferedIterator<T> {
-  private _sources : InternalSource<T>[] = [];
+  private _sources : {read: boolean, source: InternalSource<T>}[] = [];
```

Comment:
```suggestion
  private _sources : { read: boolean, source: InternalSource<T> }[] = [];
```
weird that the linter didn't whine. It should. Something wrong with CI?